tf2onnx/rewriter/rnn_utils.py

```diff
@@ -554,27 +570,41 @@ def belong_to_birnn(g, fw_rnn, bw_rnn, rnn_type):
     return True
 
 
-def get_reverse_nodes_after_y_output(g, rnn_bw):
+def is_tail_slice_op(node):
+    return (
+        node.type == 'StridedSlice' and
+        node.inputs[1].get_tensor_value() == [-1] and
+        node.inputs[2].get_tensor_value() == [0] and
+        node.inputs[3].get_tensor_value() == [1] and
+        node.get_attr('shrink_axis_mask').i == 1
+    )
+
+
+def get_reverse_or_slice_nodes_after_y_output(g, rnn_bw):
     bw_consumers = g.find_output_consumers(rnn_bw.output[0])
 
     # todo: figure out a better way to remove reverse op
     squeeze_nodes = [c for c in bw_consumers if c.type == "Squeeze"]
     s_cnt = len(squeeze_nodes)
     if s_cnt == 1:
         s = squeeze_nodes[0]
-        trans_nodes = g.find_output_consumers(s.output[0])
-        if len(trans_nodes) == 1:
-            if trans_nodes[0].type == "Transpose":
-                reverse_nodes = g.find_output_consumers(trans_nodes[0].output[0])
-            elif utils.is_tf_reverse_op(trans_nodes[0]):
-                reverse_nodes = trans_nodes
-            else:
-                logger.debug("not found reverse op, unexpected")
-                return []
-
-            are_all_reverse = all([utils.is_tf_reverse_op(r_op) for r_op in reverse_nodes])
-            if are_all_reverse:
-                return reverse_nodes
+        reverse_or_slice_nodes = g.find_output_consumers(s.output[0])
+        if len(reverse_or_slice_nodes) == 1:
+            if reverse_or_slice_nodes[0].type == "Transpose":
+                reverse_or_slice_nodes = g.find_output_consumers(reverse_or_slice_nodes[0].output[0])
+
+            if len(reverse_or_slice_nodes) == 1 and reverse_or_slice_nodes[0].type == "Identity":
+                reverse_or_slice_nodes = g.find_output_consumers(reverse_or_slice_nodes[0].output[0])
```

Comment:
Should we do this twice here? Or update it to :

            if len(reverse_or_slice_nodes) == 1 and reverse_or_slice_nodes[0].type == "Identity":
                reverse_or_slice_nodes = g.find_output_consumers(reverse_or_slice_nodes[0].output[0])
                if len(reverse_or_slice_nodes) == 1 and reverse_or_slice_nodes[0].type == "Identity":
                    reverse_or_slice_nodes = g.find_output_consumers(reverse_or_slice_nodes[0].output[0])